api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1923,4 +1923,23 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
 	
+	/**
+	 * Get the concept by conceptRef where the conceptRef can either be: 1) an integer id like 5090
+	 * 2) a mapping type id like "XYZ:HT" 3) a uuid like "a3e12268-74bf-11df-9768-17cfc9833272" 4) a
+	 * name like "PLATELETS" 5) the fully qualified name of a Java constant that contains one of
+	 * above
+	 * 
+	 * @param conceptRef the concept string identifier
+	 * @return the concept if exist, else null
+	 * <strong>Should</strong> find a concept by its conceptId
+	 * <strong>Should</strong>  find a concept by its mapping
+	 * <strong>Should</strong> find a concept by its uuid
+	 * <strong>Should</strong> find a concept by its name
+	 * <strong>Should</strong> find a concept by a fully qualified name of a Java constant that contains an id,
+	 * 								uuid, name or mapping
+	 * <strong>Should</strong> return null when ConceptRef does not exist
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public Concept getConceptByNameOrIdOrMap(String conceptRef);
```

Comment:
@jnsereko thanks for the work on this, i think the method name should be` getConcept()` and the docs above plus the added unit tests will explain more about what the method does

Reply:
there is an aready existing method called `getConcept(String)` that gets a concept by id and name

Reply:
oh okay,,,,,,,,,,
kindly have a look at the travis ci failure and also don't forget to squash the commits